include/deal.II/distributed/cell_weights.h

```diff
@@ -38,80 +38,178 @@ namespace parallel
    *
    * @ingroup distributed
-   * @author Marc Fehling, 2018
+   * @author Marc Fehling, 2018, 2019
    */
   template <int dim, int spacedim = dim>
   class CellWeights
   {
   public:
+    /**
+     * An alias that defines the characteristics of a function that can be used
+     * for weighting cells during load balancing.
+     *
+     * Such weighting functions take as arguments an iterator to a cell and the
+     * future finite element that will be assigned to it after repartitioning.
+     * They return an unsigned integer which is interpreted as the cell's
+     * weight or, in other words, the additional computational load associated
+     * with it.
+     */
+    using WeightingFunction = std::function<unsigned int(
+      const typename hp::DoFHandler<dim, spacedim>::cell_iterator &,
+      const FiniteElement<dim, spacedim> &)>;
+
     /**
      * Constructor.
      *
      * @param[in] dof_handler The hp::DoFHandler which will be used to
      *    determine each cell's finite element.
+     * @param[in] weighting_function The function that determines each
+     *    cell's weight during load balancing.
      */
-    CellWeights(const dealii::hp::DoFHandler<dim, spacedim> &dof_handler);
+    CellWeights(const dealii::hp::DoFHandler<dim, spacedim> &dof_handler,
+                const WeightingFunction &weighting_function);
 
     /**
      * Destructor.
+     *
+     * Disconnects the function previously connected to the weighting signal.
      */
     ~CellWeights();
 
     /**
-     * Choose a constant weight @p factor on each cell.
+     * Connect a different @p weighting_function to the Triangulation
+     * associated with the @p dof_handler.
+     *
+     * Disconnects the function previously connected to the weighting signal.
      */
     void
+    reinit(const hp::DoFHandler<dim, spacedim> &dof_handler,
+           const WeightingFunction &            weighting_function);
+
+    /**
+     * Converts a @p weighting_function to a different type that qualifies as
+     * a callback function, which can be connected to a weighting signal of a
+     * Triangulation.
+     *
+     * This function does <b>not</b> connect the converted function to the
+     * Triangulation associated with the @p dof_handler.
+     */
+    static std::function<unsigned int(
+      const typename Triangulation<dim, spacedim>::cell_iterator &cell,
+      const typename Triangulation<dim, spacedim>::CellStatus     status)>
+    make_weighting_callback(const hp::DoFHandler<dim, spacedim> &dof_handler,
+                            const WeightingFunction &weighting_function);
+
+    /**
+     * @name Selection of weighting functions
+     * @{
+     */
+
+    /**
+     * Choose a constant weight @p factor on each cell.
+     */
+    static WeightingFunction
+    constant_weighting(const unsigned int factor = 1000);
+
+    /**
+     * The pair of floating point numbers \f$(a,b)\f$ provided via
```

Comment:
Here and below, remove the `\f` -- we have a script that allows us to write just `$...$`.
```suggestion
     * The pair of floating point numbers $(a,b)$ provided via
```